webauthn/Sources/App/Controllers/WebAuthnController.swift

```diff
@@ -20,167 +20,126 @@ import HummingbirdCore
 import WebAuthn
 
 struct HBWebAuthnController {
+    let webauthn: WebAuthnManager
+
     func add(_ group: HBRouterGroup) {
         group
-            .post("signup", options: .editResponse, use: SignupHandler.self)
+            .post("signup", options: .editResponse, use: self.signin)
             .get("login", options: .editResponse, use: self.beginAuthentication)
+        group
+            .add(middleware: WebAuthnSessionStateAuthenticator())
+            .post("beginregister", use: self.beginRegistration)
+            .post("finishregister", use: self.finishRegistration)
+            .post("login", options: .editResponse, use: self.finishAuthentication)
         group
             .add(middleware: WebAuthnSessionAuthenticator())
-            .post("beginregister", use: BeginRegistrationHandler.self)
-            .post("finishregister", use: FinishRegistrationHandler.self)
-            .post("login", options: .editResponse, use: FinishAuthenticationHandler.self)
-            .get("test", use: self.authenticationDetails)
+            .get("test", use: self.getUser)
     }
 
-    struct SignupHandler: HBAsyncRouteHandler {
-        struct Input: Decodable {
-            let name: String
-        }
-
-        typealias Output = HBResponse
-
-        let input: Input
-
-        init(from request: HBRequest) throws {
-            self.input = try request.decode(as: Input.self)
-        }
+    struct SignInInput: Decodable {
+        let name: String
```

Comment:
```suggestion
        let username: String
```

Reply:
username, name I keep swapping between the two. I've changed this one as well